relay-spans/src/v2_to_v1.rs

```diff
@@ -270,6 +277,178 @@ fn derive_op_for_v2_span(span: &SpanV2) -> String {
     op
 }
 
+/// Generates a `sentry.description` attribute for V2 span, if possible.
+///
+/// This uses attributes of the span to figure out an appropriate description, trying to match what
+/// the SDK might have sent. This works well for HTTP and database spans, but doesn't have a
+/// thorough implementation for other types of spans for now.
+fn derive_description_for_v2_span(span: &SpanV2) -> Option<String> {
+    // `name` is a low-cardinality description of the span, so it makes for a good fallback.
+    let description = span.name.value().map(|v| v.to_owned());
+
+    let Some(attributes) = span.attributes.value() else {
+        return description;
+    };
+
+    if attributes.contains_key("http.request.method") || attributes.contains_key("http.method") {
+        if let Some(http_description) = derive_http_description(attributes, &span.kind.value()) {
+            return Some(http_description);
+        }
+    }
+
+    if attributes.contains_key("db.system") || attributes.contains_key("db.system.name") {
+        let is_cache_op = attributes
+            .get("sentry.op")
+            .and_then(|attr| attr.value())
+            .and_then(|attr_val| attr_val.value.value.value())
+            .and_then(|v| v.as_str())
```

Comment:
This pattern is starting to show up a _lot_. Maybe worth adding a method like `dig` to `Attributes` to pull this data out?